nixos/modules/virtualisation/lxc-container.nix

```diff
@@ -1,26 +1,137 @@
-{ lib, ... }:
+{ lib, config, pkgs, ... }:
 
 with lib;
 
+let
+  templateSubmodule = { ... }: {
+    options = {
+      enable = mkEnableOption "this template";
+
+      target = mkOption {
+        description = "Path in the container";
+        type = types.path;
+      };
+      template = mkOption {
+        description = ".tpl file for rendering the target";
+        type = types.path;
+      };
+      when = mkOption {
+        description = "Events which trigger a rewrite (create, copy)";
+        type = types.listOf (types.str);
+      };
+      properties = mkOption {
+        description = "Additional properties";
+        type = types.attrs;
+        default = {};
+      };
+    };
+  };
+
+  toYAML = name: attrs: pkgs.runCommandNoCC name {
```

Comment:
Why does it have to be YAML? if the tool is fine with JSON we can skip the local build and the requirement of having remarshal locally. I find it annoying very frequently with other modules.

Reply:
Also if you want to do it this way remove the indirection through `writeFile` and use `passAsFile` instead. https://nixos.org/manual/nix/stable/#sec-advanced-attributes

Reply:
I'd have to see if they support json, so far I've only seen yaml for metadata used

Reply:
YAML is a superset of JSON, so if they support parsing yaml, giving json should be fine.

Reply:
This is based on the assumption the same code can parse both JSON and YAML, which is not the case.